Source/Scene/Cesium3DTileset.js

```diff
@@ -1755,14 +1911,110 @@ define([
+             * of an object, they will always be drawn while loading, even if backface culling is enabled.
+             */
+
+            commandList.push(stencilClearCommnad);
+
+            var command, rs;
+
+            var initialLength = commandList.length;
+            backfaceCommands.length = 0;
+
+            for (i = 0; i < length; ++i) {
+                tile = selectedTiles[i];
+                if (tile.selected) {
+                    // Raise visible event before update in case the visible event
+                    // makes changes that update needs to apply to WebGL resources
+                    tileVisible.raiseEvent(tile);
+                    incrementPointAndFeatureSelectionCounts(tileset, tile.content);
+
+                    var lengthBeforeUpdate = commandList.length;
+                    tile.update(tileset, frameState);
+                    for (var j = lengthBeforeUpdate; j < commandList.length; ++j) {
+                        command = DrawCommand.shallowClone(commandList[j]);
+                        commandList[j] = command;
+
+                        if (!tile._finalResolution) {
+                            // draw backfaces of all unresolved tiles so resolved tiles don't poke through
+                            if (command.renderState.depthMask) {
+                                backfaceCommands.push(DrawCommand.shallowClone(command));
+                            }
+                        }
+
+                        // ignore if tile does not write depth (ex. translucent)
+                        if (command.renderState.depthMask) {
+                            // Tiles only draw if their selection depth is >= the tile drawn already. They write their
+                            // selection depth to the stencil buffer to prevent ancestor tiles from drawing on top
+                            rs = clone(command.renderState, true);
+                            rs.stencilTest.enabled = true;
+                            rs.stencilTest.reference = tile._selectionDepth;
+                            rs.stencilTest.frontFunction = StencilFunction.GREATER_OR_EQUAL;
+                            rs.stencilTest.frontOperation.zPass = StencilOperation.REPLACE;
+                            command.renderState = RenderState.fromCache(rs);
+                        }
+                    }
+                }
+            }
+
+            var addedCommandsLength = commandList.length - initialLength;
+            var backfaceCommandsLength = backfaceCommands.length;
+
+            // write just depth of unresolved tiles so resolved final res tiles do not appear in front
+            for (i = 0; i < backfaceCommandsLength; ++i) {
+                command = backfaceCommands[i];
+                rs = clone(command.renderState, true);
+                rs.cull.enabled = true;
+                rs.cull.face = CullFace.FRONT;
+                command.renderState = RenderState.fromCache(rs);
```

Comment:
I think I left this on for the cases you're inside an object and still want to see color. The backfaces would get drawn twice (once for the depth pass and once for the draw pass with stencil), but the second time with color won't draw anything because the depth test is LESS_THAN and not LESS_THAN_OR_EQUAL.

1. Change the depth to be <=?
2. Leave the color mask if it's enabled, but if it's not use a separate depth-only fragment shader?